packages/desktop/views/dashboard/drawers/dapp-config/views/ConnectionRequestDrawer.svelte

```diff
@@ -27,9 +27,17 @@
     $: isVerified = $sessionProposal?.verifyContext.verified.validation === SessionVerification.Valid
     $: alreadyConnected = !!getPersistedDappNamespacesForDapp($sessionProposal?.params.proposer.metadata.url)
     $: unsupportedMethods = $sessionProposal ? getUnsupportedMethods($sessionProposal?.params.requiredNamespaces) : []
-    $: unsupportedNetworks = $sessionProposal ? getUnsupportedNetworks($sessionProposal?.params.requiredNamespaces) : []
-    $: isSupportedOnOtherProfiles = $sessionProposal ? areNetworksSupportedOnOtherProfiles(unsupportedNetworks) : false
-    $: fulfillsRequirements = unsupportedMethods.length === 0 && unsupportedNetworks.length === 0
+    $: supportedNetworks = $sessionProposal
+        ? getSupportedNetworks($sessionProposal?.params.requiredNamespaces, $sessionProposal?.params.optionalNamespaces)
+        : []
+    $: unsupportedRequiredNetworks = $sessionProposal
+        ? getUnsupportedRequiredNetworks($sessionProposal?.params.requiredNamespaces)
+        : []
+    $: isSupportedOnOtherProfiles = $sessionProposal
+        ? areNetworksSupportedOnOtherProfiles(unsupportedRequiredNetworks)
+        : false
```

Comment:
It's better to move the `$sessionProposal` checks into the respective functions to simplify the arguments for l.31 and for cleaner code.